vita3k/io/src/io.cpp

```diff
@@ -482,13 +490,15 @@ SceUID read_dir(IOState &io, const SceUID fd, emu::SceIoDirent *dent, const std:
             if (fs::is_regular_file(cur_path)) {
                 if (stat_file(io, file_path.c_str(), &dent->d_stat, pref_path, base_tick, export_name) < 0)
                     return IO_ERROR(SCE_ERROR_ERRNO_EMFILE);
+                else
+                    return 1; // move to the next file
             }
             if (fs::is_directory(cur_path)) {
-                const auto dir_fd = open_dir(io, file_path.c_str(), pref_path, export_name);
-                return read_dir(io, dir_fd, dent, pref_path, base_tick, export_name);
+                stat_file(io, file_path.c_str(), &dent->d_stat, pref_path, base_tick, export_name);
```

Comment:
it looks like this identical line is executed if either `is_regular_file` or `is_directory` so why check?

Reply:
huh, you're right. i had the old code commented while testing it so i didn't notice lol